### ISSUE TITLE
core.setFailed should accept Errors

### ISSUE THREAD

**Issue by jasonkarns** [actions/toolkit#386]
**Describe the enhancement**
It's extremely common (nay, the _purpose_) of `core.setFailed` to report Errors that are caught during an action. And yet, its method signature does not accept `Error` instances.

It would be beneficial to accept Error instances instead of requiring users to convert to strings.

**Code Snippet**
This would be _especially_ beneficial for promise chains:

```js
somePromise()
  .then(doThing)
  .then(doOther)
  .then(whatever)
  .catch(core.setFailed)
```

Present behavior:

`core.setFailed(Error("foo"))` results in: `TypeError: (s || "").replace is not a function`

**Comment by thboop** [comment]
I agree.

**Comment by bnb** [comment]
+1. Putting the burden on users to pass the proper type rather than handling coercion in the module is not good DX.

**Comment by thboop** [comment]
Resolved in #405
